pkg/probes/curl/curl_json.go

```diff
@@ -81,8 +77,8 @@ func (clp Probe) GetMachineImageID(platformType string, cpuArch cpu.Architecture
 // variables listed in the template's "network-verifier-required-variables" directive, or if
 // values *are* provided for variables that must be set to a certain value for the probe to
 // function correctly (presetUserDataVariables) -- this function will fill-in those values for you.
-func (clp Probe) GetExpandedUserData(userDataVariables map[string]string) (string, error) {
-	// Extract required variables specified in template (if any)
+func (clp Probe) GetExpandedUserData(userDataVariables map[string]string, userDataTemplate string) (string, error) {
+	// Extract required variables specified in template (if any) based off platformType
```

Comment:
Does the behavior here actually depend on `platformType`?

Reply:
It does not! I have removed part of that comment.